pyproject.toml

```diff
@@ -2,13 +2,31 @@
 requires = ['setuptools', 'wheel', 'requests']
 build-backend = 'setuptools.build_meta'
 
-[tool.black]
+[tool.ruff]
+target-version = "py38"
 line-length = 125
 
-[tool.isort]
-no_sections = true
-force_alphabetical_sort = true
-combine_as_imports = true
-profile = "black"
-skip_gitignore = true
-line_length = 125
+[tool.ruff.lint]
+select = [
```

Comment:
suggestion: I would consider also enabling these. No opinion though as to whether they should be included in this initial introduction. (Copied from another project, with some amended comments).

```toml
extend-select = [
  # bugbear
  "B",
  # comprehensions
  "C4",
  # mccabe
  "C90",
  # bandit -- finds security issues, I ignore S101 when using this. Author of pre-commit summarizes
  # reasoning for that well here: https://stackoverflow.com/a/68429294/1220706.
  "S",
  # blind exception
  # Bare excepts are caught without this, but this also catches `except Exception: ...`.
  "BLE",
  # prevents shadowing builtins
  "A",
  # pycodestyle
  "W",
  # pyupgrade -- automatically upgrade to modern syntax
  "UP",
  # debugger
  "T10",
  # print -- might not be relevant
  "T20",
  # quotes
  "Q",
  # flake8-return
  # RET is theoretically nice, but there were too many false positives for it to be useful when I tried it.
  "RET",
  # simplify
  "SIM",
  # tidy imports
  # We use this to only outlaw relative parent imports.
  "TID",
]

[tool.ruff.mccabe]
max-complexity = 10

[tool.ruff.flake8-tidy-imports]
ban-relative-imports = "parents"
```

Reply:
I prefer to start small, but I added some and fixed the associated violations.

Reply:
Makes sense! 👍